src/foundation.cs

```diff
@@ -8258,6 +8258,12 @@ partial interface NSUrlSessionConfiguration : NSCopying {
 		[MacCatalyst (16, 0)]
 		[Export ("requiresDNSSECValidation")]
 		bool RequiresDnsSecValidation { get; set; }
+
+		// not yet ready, waigin on pr https://github.com/xamarin/xamarin-macios/pull/7539 to be fixed
+		// to land this and the manual pro[erty
+		// [Internal]
+		// [Export ("proxyConfigurations", ArgumentSemantic.Copy)]
```

Comment:
This is defined as an [`NSArray<nw_proxy_config_t>`](https://developer.apple.com/documentation/foundation/nsurlsessionconfiguration/4172960-proxyconfigurations), where `nw_proxy_config_t` is defined as an NSObject that implements the [OS_nw_proxy_config_t](https://developer.apple.com/documentation/network/os_nw_proxy_config?language=objc) protocol: https://developer.apple.com/documentation/network/nw_proxy_config_t?language=objc.

This means that you don't need Dustin's PR.